### DISCUSSION TITLE
How to display collection field choices and keep them dynamically synced

### DISCUSSION THREAD

**Discussion by kuwts** [pocketbase/pocketbase#5565] [Q&A]
I am trying to get the choices for a specific field so that I can add them into a form and keep it synced with PB

```
// Fetch the collection schema
const businessesCollection = await pb.collections.getOne('businesses');

// Extract all language options
const languages = businessesCollection.schema.fields.find(
  field => field.name === 'language'
)?.options?.values || [];

// Extract all category options
const categories = businessesCollection.schema.fields.find(
  field => field.name === 'category'
)?.options?.values || [];

// Extract all country options
const countries = businessesCollection.schema.fields.find(
  field => field.name === 'countries'
)?.options?.values || [];
```

however this is giving the error 

```12:24:12 [ERROR] The request requires valid admin authorization token to be set.```

I saw another issue where they advised making a separate collection for each option, but that seems a bit excessive and it was a few years ago so maybe there have been changes to this since then. Do you guys know how to get the options that users can choose from dynamically or is there a simpler approach that I am not thinking of?

**Comment by international-omelette** [comment]
I don‘t know if there‘s a better way to get option values now but it seems to be totally ok to give categories, countries and languages their own collections. 
You never know if someday you’d like to add more fields to these option sets and there are no downsides in having many tables.

**Reply by kuwts** [reply]
> I don‘t know if there‘s a better way to get option values now but it seems to be totally ok to give categories, countries and languages their own collections.
> You never know if someday you’d like to add more fields to these option sets and there are no downsides in having many tables.

Thanks! How would I then reference it in Pocketbase? For example, if I had a countries table, how could I then add that as options to the field in the businesses collection?

**Reply by international-omelette** [reply]
I think what you are looking for is a relation field: https://pocketbase.io/docs/working-with-relations/
If this is not what you are looking for, please elaborate on your use case.

**Comment by dschissler** [comment]
What is your framework type?  Is PB just a API server or do you have access to JS or Golang to make custom routes?  I can point you at the documentation and such, but first tell me about your setup.
